src/components/Accordion/PAccordion.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="p-accordion">
+    <template v-for="section in sections" :key="section">
+      <div class="p-accordion__section">
+        <div class="p-accordion__header">
+          <slot :name="`${section}-heading`">
+            <slot name="heading" :section="section">
+              {{ section }}
+            </slot>
+          </slot>
+          <p-button
+            size="sm"
+            icon="ChevronDownIcon"
+            flat
+            rounded
+            class="p-accordion__button"
+            :class="getButtonClass(String(section))"
+            @click="toggleSection(String(section))"
+          />
+        </div>
+
+        <PAutoHeightTransition>
+          <template v-if="isOpen(String(section))">
+            <div class="p-accordion__content">
+              <!-- this is needed because PAutoHeightTransition does not animate an elements padding correctly -->
+              <div class="p-accordion__content-padding">
+                <slot :name="`${section}-content`">
+                  <slot name="content" :section="section" />
+                </slot>
+              </div>
+            </div>
+          </template>
+        </PAutoHeightTransition>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup generic="const T extends Readonly<string[]>">
+  import { Ref, ref } from 'vue'
+  import { PAutoHeightTransition } from '@/components/AutoHeightTransition'
+
+  const props = defineProps<{
+    sections: T,
+  }>()
+
+  const open: Ref<T[number] | null> = ref(props.sections[0])
+
+  function toggleSection(section: string): void {
+    if (isOpen(section)) {
+      open.value = null
+      return
+    }
+
+    open.value = section
+  }
+
+  function isOpen(section: string): boolean {
+    return open.value === section
+  }
+
+  function getButtonClass(section: string): string | undefined {
```

Comment:
Elsewhere we compute a list of conditional classes, nbd. A little surprised this is recalled when the state changes.

Reply:
Yeah frequently you'll see a computed for something like this. But when you're getting dynamic classes for elements in a v-for this is usually how that's done. Vue is pretty smart about re-evaluating method calls that are referenced in the template and I've never had an issue with this. Computes are preferred but in a v-for a method is just much simpler. Otherwise we'd have to make a computed with dynamic keys or something.